README
 gastify , Jan Penschuck, penschck@gmail.com
	

 gastify is a simple notification client for the asterisk plugin app_notify.
 It sits in background and notifies you about an incoming call and displays
 number and name of the caller with a simple popup.
 
 Features: 	* call notification
 		* call history

 gastify is loosely hacked up with the help of http://www.debian-administration.org/articles/407,
 http://systhread.net/texts/200506cdaemon1.php, liferea sourcecode and the sampleClient of the app_notify author
 this is my first c app. but i will try to produce a good code. any suggestions and bugreports are welcome. 


 Requirements: 	glib2, gtk2, libnotify
 
 License: 	GPL2

 Author:	Jan Penschuck


ChangeLog
2009-11-06 penschuck  <penschuck@gmail.com>

	* released v1.3.0

2009-11-04 penschuck  <penschuck@gmail.com>

	* changed date in listview
	* added missing copy icon to context menu
	* reverted r65
	* added Alessandro Zummo to AUTHORS

2009-10-12 penschuck  <penschuck@gmail.com>

	* ui should now be Gnome HIG compatible

2009-10-09 penschuck  <penschuck@gmail.com>

	* small changes in gastify ui to hide callbar until -c is given
	* some callbar ui rework
	* bump to autotools 1.11
	* typo fixes

2009-10-06 penschuck  <penschuck@gmail.com>

	* removed callscript.php and added a bash script to place a call

2009-09-24 penschuck  <penschuck@gmail.com>

        * reworked gettext files
        * fixed --help translation
	* some fixes in gui.c
	* added extras folder
	* added php script to call a number using asterisk manager

2009-09-27  gettextize  <bug-gnu-gettext@gnu.org>

	* m4/gettext.m4: New file, from gettext-0.17.
	* m4/iconv.m4: New file, from gettext-0.17.
	* m4/lib-ld.m4: New file, from gettext-0.17.
	* m4/lib-link.m4: New file, from gettext-0.17.
	* m4/lib-prefix.m4: New file, from gettext-0.17.
	* m4/nls.m4: New file, from gettext-0.17.
	* m4/po.m4: New file, from gettext-0.17.
	* m4/progtest.m4: New file, from gettext-0.17.

2009-09-24 penschuck  <penschuck@gmail.com>

	* fixed glade translation markups
	* fixed de translation

2009-09-23 azummo <azummo-google@towertech.it>

	* added better pixmaps	

2009-09-22 azummo <azummo-google@towertech.it>

	* the cid list is now a tree view with
	* copy to clipboard and google search options.
	* added --call-cmd, --strip-leading
	* the call command is triggered by a double click on the
	* treeview or by the quick call text entry field.

2009-01-06  penschuck  <penschuck@gmail.com>

    * dropped glade dep in favor of GtkBuilder
    * fixed wrong init of GOptionEntry

2008-09-24  penschuck  <penschuck@gmail.com>

    * small change in log-format
    * moved checkinstall.sh buildpackages.sh
    * buildpackage.sh produces tarball too
    * released v1.2.3

2008-09-21  penschuck  <penschuck@gmail.com>

    * switched license to LGPL3
    * fixed --prefix at configure time
    * small change in checkinstall.sh
    * bumb to automake-1.10

2008-09-19  penschuck  <penschuck@gmail.com>

    * translation updates
    * added script that generates a deb and a rpm: checkinstall.sh

2008-09-16  penschuck  <penschuck@gmail.com>

    * added ability to execute a cmd when new call arrives (-e)
    * fixed --help, but still throws warning at compile time :/

2008-09-15  penschuck  <penschuck@gmail.com>

    * added ability to disable notifications in statusicon-menu

2007-07-20  penschuck  <penschuck@gmail.com>

    * compiles now with -Wall -O -W -Wshadow -pedantic

2007-07-01  penschuck  <penschuck@gmail.com>

    * 1.2.2 release
   
2007-06-28  penschuck  <penschuck@gmail.com>

    * some about window fixes 

2007-06-28  penschuck  <penschuck@gmail.com>

    * more strcpy and strcat fixes

2007-06-28  penschuck  <penschuck@gmail.com>

    * removed linebreak in notification text

2007-06-28  penschuck  <penschuck@gmail.com>

    * gettext fixes

2007-06-23  penschuck  <penschuck@gmail.com>

    * 1.2.1 release
    
2007-06-21  penschuck  <penschuck@gmail.com>

    * replaced strcpy and strcat with g_strconcat (more secure)
    
2007-06-21  penschuck  <penschuck@gmail.com>

    * sort order in history window changed to: latest call on top now
    
2007-05-12  penschuck  <penschuck@gmail.com>

    * code cleanups

2007-03-10  penschuck  <penschuck@gmail.com>

    * 1.2 release

2007-03-10  penschuck  <penschuck@gmail.com>

    * added statusicon change on call

2007-03-10  penschuck  <penschuck@gmail.com>

    * small pkg-config fix for glade in Makefie.am

2006-11-02  penschuck  <penschuck@gmail.com>

    * better icon

2006-10-07  penschuck  <penschuck@gmail.com>

    * 1.1 release

2006-10-05  penschuck  <penschuck@gmail.com>

    * changes how the app icon is handled 

2006-10-03  penschuck  <penschuck@gmail.com>

    * added scrollbars to log window

2006-10-03  penschuck  <penschuck@gmail.com>

    * fixed bug with history window-close
	
2006-10-02  penschuck  <penschuck@gmail.com>

    * added german translation

2006-10-02  gettextize  <bug-gnu-gettext@gnu.org>

    * Makefile.am (SUBDIRS): Add po.
    (ACLOCAL_AMFLAGS): Add -I m4.
    (EXTRA_DIST): Add config.rpath, m4/ChangeLog.
    * configure.in (AC_OUTPUT): Add po/Makefile.in.
	
2006-09-29  penschuck  <penschuck@gmail.com>

    * code reconstruction

2006-09-29  penschuck  <penschuck@gmail.com>

    * replaced my ugly options parser with the glib commandline option parser

2006-09-28  penschuck  <penschuck@gmail.com>

    * added Homepage to the about dialog

2006-09-28  penschuck  <penschuck@gmail.com>

    * added timestamps to history

2006-09-28  penschuck  <penschuck@gmail.com>

    * click again on trayicon closes history window now


2006-09-26  penschuck  <penschuck@gmail.com>

    * initial 1.0 release


trunk/src/main.c
/* Mi Jul 26 17:50:21 UTC 2006
 *
 * Copyright (C) 2006-2008 by Jan Penschuck                                                    
 *
 * This file is part of gastify.
 *
 * gastify is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * gastify is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with gastify.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

#include <config.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>
#include <sys/stat.h>
#include <locale.h>
#include <glib.h>
#include <glib/gi18n.h>
#include <gtk/gtk.h>
#include "socket.h"
#include "gui.h"
	
static gint port = 40000;
static gboolean daemonize = FALSE;
static gboolean version = FALSE;

gint  popuptime = 10;
gchar *execcmd = NULL;
gchar *callcmd = NULL;
gboolean strip_leading = FALSE;

static GOptionEntry entries[] = 
{
	{ "port", 'p', 0, G_OPTION_ARG_INT, &port, "Listen on given port", "N" },
	{ "time", 't', 0, G_OPTION_ARG_INT, &popuptime, "Display popup given time", "SECONDS" },
	{ "demonize", 'd', 0, G_OPTION_ARG_NONE, &daemonize, "Forks in the background", NULL },
	{ "execute", 'e', 0, G_OPTION_ARG_STRING, &execcmd, "Executes command on new call", "COMMAND TO EXECUTE" },
	{ "call-cmd", 'c', 0, G_OPTION_ARG_STRING, &callcmd, "Command to execute to initiate a call", "COMMAND" },
	{ "strip-leading", 's', 0, G_OPTION_ARG_NONE, &strip_leading, "Strip the leading digit on the received CID", NULL },
	{ "version", 'V', 0, G_OPTION_ARG_NONE, &version, "Print version info", NULL },	
	{ NULL, 0, 0, 0, NULL, NULL, NULL }
};

int main(int argc, char *argv[])
{	
	GOptionContext *context;
	
	/* gettext */
  	setlocale(LC_ALL, "");
  	bindtextdomain(GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR);
	textdomain(GETTEXT_PACKAGE);
	
	/* commandline options */
	context = g_option_context_new("- Gastify Call Notification");
	g_option_context_set_translation_domain(context, GETTEXT_PACKAGE);	
 	g_option_context_add_main_entries(context, entries, GETTEXT_PACKAGE);
	g_option_context_add_group(context, gtk_get_option_group(TRUE));
	g_option_context_parse(context, &argc, &argv, NULL);
	
	
	/* print version info */		
	if (version) {
		printf("%s %s\n", "Version", VERSION);
		exit(0);		
	}

	/* demonize or not */
	if (daemonize) {				
		pid_t pid, sid;
		pid = fork();
	
		if (pid < 0) {
			exit(EXIT_FAILURE);
		} 
		else if (pid > 0) {
			exit(EXIT_SUCCESS);
		}

		umask(0);
		sid = setsid();
		if (sid < 0) {
			exit(EXIT_FAILURE);
		}						
	}	
	
	g_option_context_free(context);
	
	/* get up and running */
	g_set_prgname(PACKAGE);
	gtk_window_set_default_icon_name(PACKAGE);
  	gtk_set_locale();
	gtk_init(&argc, &argv);
	
	/* open socket and initialize statusicon */
	watchSocket(port, initializeGui());
	gtk_main();
	
	return 0;
}
